src/metabase/pulse.clj

```diff
@@ -261,24 +264,35 @@
          :email (:email recipient)}
         {:kind :user
          :user recipient}))
+    :http
+    []
     (do
      (log/warnf "Unknown channel type %s" (:channel_type pulse-channel))
      [])))
 
-(defn- channel-send!
-  [& args]
-  (try
-    (apply channel/send! args)
-    (catch Exception e
-      ;; Token errors have already been logged and we should not retry.
-      (when-not (and (= :channel/slack (:type (first args)))
-                     (contains? (:errors (ex-data e)) :slack-token))
-        (throw e)))))
-
 (defn- send-retrying!
-  [& args]
+  [pulse-id channel message]
   (try
-    (apply (retry/decorate channel-send!) args)
+    (let [;; once we upgraded to retry 2.x, we can use (.. retry getMetrics getNumberOfTotalCalls) instead of tracking
+          ;; this manually
+          retry-count (volatile! 0)
+          send!       (fn []
+                          (task-history/with-task-history {:task         "channel-send"
```

Comment:
On the choice of "channel-send": the task name is user-facing and as far as I know channel isn't in a user's vocabulary yet. Would "send-notification" be more appropriate as a user-facing name for this task?